### DISCUSSION TITLE
[@types/node] Readable::on callback signature includes any

### DISCUSSION THREAD

**Discussion by mlntdrv** [DefinitelyTyped/DefinitelyTyped#55838] [Issues with a @types package]
Using the native HTTP module with DT types.
One of the _Readable::on_ overloads in _@types/node/stream.d.ts_ is:
```ts
on(event: "data", listener: (chunk: any) => void): this;
```
This `chunk: any` is not typesafe because it didn't force me to do any type checks or validation before operating on it.
Please type it with a narrower type, or if it is not possible, as `unknown`.
![chunk-any](https://user-images.githubusercontent.com/44335787/133714366-c6d17048-99d6-4afa-9237-6edfb51c9187.PNG)

[a comment by an automated account is not shown]

**Comment by SimonSchick** [comment]
This is a current restriction of the either the event emitter pattern in node or the fact that TS does not inherit method overloads.
